CRM/Utils/Check/Component/Schema.php

```diff
@@ -88,4 +88,76 @@ public function checkMissingLogTables() {
     return $messages;
   }
 
+  /**
+   * @return array
+   */
+  public function checkSmartGroupCustomFieldCriteria() {
+    $messages = $problematicSG = [];
+    $customFieldIds = array_keys(CRM_Core_BAO_CustomField::getFields('ANY', FALSE, FALSE, NULL, NULL, FALSE, FALSE, FALSE));
+    $smartGroups = civicrm_api3('SavedSearch', 'get', [
+      'sequential' => 1,
+      'options' => ['limit' => 0],
+    ]);
+    if (empty($smartGroups['values'])) {
+      return $messages;
+    }
+    foreach ($smartGroups['values'] as $group) {
+      if (empty($group['form_values'])) {
+        continue;
+      }
+      foreach ($group['form_values'] as $formValues) {
+        if (substr($formValues[0], 0, 7) == 'custom_') {
+          list(, $customFieldID) = explode('custom_', $formValues[0]);
+          if (!in_array($customFieldID, $customFieldIds)) {
+            $problematicSG[CRM_Contact_BAO_SavedSearch::getName($group['id'], 'id')] = [
+              'title' => CRM_Contact_BAO_SavedSearch::getName($group['id'], 'title'),
+              'cfid' => $customFieldID,
+            ];
+          }
+        }
+      }
+    }
+
+    if (!empty($problematicSG)) {
+      $html = '';
+      foreach ($problematicSG as $id => $field) {
+        if (!empty($field['cfid'])) {
+          try {
+            $customField = civicrm_api3('CustomField', 'getsingle', [
+              'sequential' => 1,
+              'id' => $field['cfid'],
+            ]);
+            $fieldName = ts('<a href="%1"> %2 </a>', [
+              1 => CRM_Utils_System::url('civicrm/admin/custom/group/field/update',
+                "action=update&reset=1&gid={$customField['custom_group_id']}&id={$field['cfid']}", TRUE
+              ),
+              2 => $customField['label'],
+            ]);
+          }
+          catch (Exception $e) {
+            $fieldName = ' <span style="color:red"> - Deleted - </span> ';
+          }
+        }
+
+        $html .= "<tr><td>{$id}</td><td>{$field['title']}</td><td class='disabled'>{$fieldName}</td>";
```

Comment:
To add edit and config group icons:

```
        $groupEdit = '<a href="' . CRM_Utils_System::url('civicrm/contact/search/advanced', "?reset=1&ssID={$id}", TRUE) . '" title="' . ts('Edit search criteria') . '"> <i class="crm-i fa-pencil"></i> </a>';
        $groupConfig = '<a href="' . CRM_Utils_System::url('civicrm/group', "?reset=1&action=update&id={$id}", TRUE) . '" title="' . ts('Group settings') . '"> <i class="crm-i fa-gear"></i> </a>';
        $html .= "<tr><td>{$id} {$groupEdit} {$groupConfig}</td><td>{$field['title']}</td><td class='disabled'>{$fieldName}</td>";
```
Aso I would convert the 'ID' and 'Group Title' and gr to one 'Group'

![proposal](https://user-images.githubusercontent.com/1064675/72544268-5241b600-3887-11ea-9529-4755c30f02f6.png)

Reply:
Note that I put on $groupConfig the group ID when should be the SSID.